bin/collect_stats.py

```diff
@@ -63,26 +87,35 @@ def main(args=None):
     microbiomes_entities_occs["entity_id"] = pd.to_numeric(microbiomes_entities_occs["entity_id"], downcast="unsigned")
 
     conditions = pd.read_csv(args.conditions, usecols=["condition_name", "microbiome_id"], sep="\t")
-    # TODO "condition_name" as " "category"?
-    # (first try: memory went up, check how to use properly)
     conditions["microbiome_id"] = pd.to_numeric(conditions["microbiome_id"], downcast="unsigned")
 
+    predictions = pd.read_csv(args.predictions, usecols=["peptide_id", "prediction_score", "allele_id"], sep="\t")
+    predictions["peptide_id"] = pd.to_numeric(predictions["peptide_id"], downcast="unsigned")
+    predictions["allele_id"] = pd.to_numeric(predictions["allele_id"], downcast="unsigned")
+    predictions["prediction_score"] = pd.to_numeric(predictions["prediction_score"], downcast="float")
+
+    alleles = pd.read_csv(args.alleles, usecols=["allele_id", "allele_name"], sep="\t").set_index("allele_id")
+    allele_names = alleles["allele_name"].to_dict()
+
+    predictions = pd.concat([df.set_index("peptide_id").rename(columns={"prediction_score":f"prediction_score_allele_{allele_id}"}).drop("allele_id", axis=1) for allele_id, df in predictions.groupby("allele_id")], join="outer", axis=1)
+    best_scored_peptides = predictions.agg("max", axis=1).rename("best_prediction_score")
+
+    stat_table = []
+
     # Process data condition-wise to reduce memory usage
     for condition_name in conditions["condition_name"]:
-        print("Process condition:", condition_name, flush=True)
-        print("Condition name:", condition_name, file=args.outfile, sep="\t", flush=True)
 
         conditions_proteins = (
             conditions[conditions.condition_name == condition_name]
             .merge(microbiomes_entities_occs)
             .drop(columns="microbiome_id")
             .merge(entities_proteins_occs)
             .drop(columns="entity_id")
+            .drop_duplicates()
```

Comment:
This would change the count in `conditions_peptides`. There might be duplicates resulting from the same protein present in multiple entities (e.g. from the download). Right?

Reply:
Oh that's absolutely true!! When following the data frames this did not occur to me, as the entities weren't present anymore! I need to drop duplicates before dropping the `entity_id` column! and then again for unique protein count.

Thanks for spotting!

Reply:
Would you expect duplicates before dropping the `entity_id` column?